bundles/org.openhab.binding.goveelan/src/main/java/org/openhab/binding/goveelan/internal/GoveeLanHandler.java

```diff
@@ -0,0 +1,390 @@
+
+            thingHandler = THING_HANDLERS.get(deviceIPAddress);
+            if (thingHandler == null) {
+                LOGGER.warn("thing Handler for {} couldn't be found.", deviceIPAddress);
+                return;
+            }
+
+            LOGGER.debug("updating status for thing {} ", thingHandler.getThing().getLabel());
+            LOGGER.trace("Response from {} = {}", deviceIPAddress, response);
+
+            if (response != null && !response.isEmpty()) {
+                try {
+                    StatusMessage statusMessage = GSON.fromJson(response, StatusMessage.class);
+                    thingHandler.updateDeviceState(statusMessage);
+                } catch (JsonSyntaxException jse) {
+                    thingHandler.updateStatus(ThingStatus.OFFLINE);
+                }
+            } else {
+                thingHandler.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/error.offline.ip-address.emptyResponse");
+            }
+            if (!thingHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+                thingHandler.updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            LOGGER.error("exception when receiving status packet {}", Arrays.toString(e.getStackTrace()));
+            // as we haven't received a packet we also don't know where it should have come from
+            // hence, we don't know which thing put offline.
+            // a way to monitor this would be to keep track in a list, which device answers we expect
+            // and supervise an expected answer within a given time but that will make the whole
+            // mechanism much more complicated and may be added in the future
+        } finally {
+            refreshJobRunning = false;
+        }
+    };
+
+    /**
+     * This thing related job <i>thingRefreshSender</i> triggers an update to the Govee device.
+     * The device sends it back to the common port and the response is
+     * then received by the common #refreshStatusReceiver
+     */
+    private final Runnable thingRefreshSender = () -> {
+        try {
+            triggerDeviceStatusRefresh();
+            if (!thing.getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/error.couldNotQueryDevice [\""
+                            + thing.getConfiguration().getProperties().get(GoveeLanConfiguration.IP_ADDRESS) + "\"]");
+        }
+    };
+
+    public GoveeLanHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        GoveeLanConfiguration goveeLanConfiguration = getConfigAs(GoveeLanConfiguration.class);
+
+        final Object ipAdress = getThing().getConfiguration().getProperties().get(GoveeLanConfiguration.IP_ADDRESS);
```

Comment:
This seems like too much work when you have just deserialized the configuration in the line above. :wink
```suggestion
        final Object ipAdress = goveeLanConfiguration.hostname;
```

First, you just need to add the missing parameters to the DTO (`GoveeLanConfiguration`).

Reply:
(also: typo - ipA**d**dress)

Reply:
omg, which actually shows a major bug because the GoveeLanConfiguration isn't really a Value class, it is only holding constants...

Reply:
I'm resolving this one since there is another comment for removing properties from the configuration definition instead. That's why they were not in the DTO.